File: TelerikUIExamples/TelerikUIExamples/DataSourceDocsQueries.m
//
//  DataSourceDocsQueries.m
//  TelerikUIExamples
//
//  Copyright © 2016 Telerik. All rights reserved.
//

#import "DataSourceDocsQueries.h"
#import "DSItem.h"
#import <TelerikUI/TelerikUI.h>

// >> datasource-predicate-style
@implementation DataSourceDocsQueries
{
    TKDataSource *dataSource;
}

-(void)viewDidLoad
{
    [super viewDidLoad];
    // >> datasource-displaykey
    NSMutableArray *items = [NSMutableArray new];
    [items addObject:[[DSItem alloc] initWithName:@"John" value:50 group:@"A"]];
    [items addObject:[[DSItem alloc] initWithName:@"Abby" value:33 group:@"A"]];
    [items addObject:[[DSItem alloc] initWithName:@"Smith" value:42 group:@"B"]];
    [items addObject:[[DSItem alloc] initWithName:@"Peter" value:28 group:@"B"]];
    [items addObject:[[DSItem alloc] initWithName:@"Paula" value:25 group:@"B"]];
    
    dataSource = [TKDataSource new];
    dataSource.itemSource = items;
    dataSource.displayKey = @"name";
    // << datasource-displaykey
    [dataSource filterWithQuery:@"value > 30"];
    [dataSource sortWithKey:@"value" ascending:true];
    [dataSource groupWithKey:@"group"];
    
    
    UITableView *tableView = [[UITableView alloc] initWithFrame:self.view.bounds];
    tableView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    tableView.dataSource = dataSource;
    [self.view addSubview:tableView];
}

@end
// << datasource-predicate-style

File: Examples/Examples/DataSource/Documentation Snippets/DataSourceDocsQueries.cs
using System;
using Foundation;
using TelerikUI;
using UIKit;

namespace Examples
{
	[Register("DataSourceDocsQueries")]
	// >> datasource-predicate-style-cs
	public class DataSourceDocsQueries : XamarinExampleViewController
	{
		TKDataSource dataSource;

		public DataSourceDocsQueries ()
		{
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			// >> datasource-displaykey-cs
			NSMutableArray items = new NSMutableArray();
			items.Add (new DSItem () { Name = "John", Value = 50, Group = "A" });
			items.Add (new DSItem () { Name = "Abby", Value = 33, Group = "A" });
			items.Add (new DSItem () { Name = "Smith", Value = 42, Group = "B" });
			items.Add (new DSItem () { Name = "Peter", Value = 28, Group = "B" });
			items.Add (new DSItem () { Name = "Paula", Value = 25, Group = "B" });

			this.dataSource = new TKDataSource ();
			this.dataSource.ItemSource = items;
			this.dataSource.DisplayKey = "Name";
			// << datasource-displaykey-cs

			this.dataSource.FilterWithQuery ("Value > 30");
			this.dataSource.SortWithKey ("Value", true);
			this.dataSource.GroupWithKey ("Group");


			UITableView tableView = new UITableView (this.View.Bounds);
			tableView.DataSource = this.dataSource;
			this.View.AddSubview (tableView);
		}
	}
	// << datasource-predicate-style-cs
}

